### DISCUSSION TITLE
Is it a good practice to nest layouts in Nuxt 3?

### DISCUSSION THREAD

**Discussion by bodrozicN** [nuxt/nuxt#26489] [Questions] [answered]
I’m working on a Nuxt 3 application and I’m considering using nested layouts for my pages. While I understand that layouts are a way to define a common structure for the pages, I’m wondering if nesting layouts (one layout inside another) is a good practice in Nuxt 3. Could this lead to unnecessary complexity or are there scenarios where this approach could simplify the structure of the application and improve its maintainability? Any insights or recommendations would be appreciated.

**Comment by aidrecabrera** [comment, marked as answer]
**TL;DR: Nested layouts depend on the project. Keep it simple and balanced. Avoid excessive nesting.**

It might sound cliche, but it depends. There are instances where they are indispensable, while in others, they may be unnecessary.

**How do you make your decision?** Well, this decision can be made prior to the implementation and development phase. That is, during the meticulous planning and design stages of your application. Evaluate your application's structure and requirements carefully to determine whether nested layouts are the optimal solution for your particular use case. You can ask yourself: _Why am I considering using nested layouts for my pages? Is it really necessary? Does it benefit the user or solve a problem?_


Peel away only what's necessary to avoid unnecessary complexity. Wasting effort on unnecessary implementations can not only be costly but may also lead to regression. So, keep it simple as possible.

Nevertheless, here's some important things to keep in mind, IMO.

- **Avoid Deep Nesting:** Limit the depth of nesting to a reasonable level. 
- **Keep it Simple:** Avoid overly complex nesting.
- **Find balance:** Strive for a balance between organization and simplicity.

That's all. What do you think? 👀
